### PR TITLE
[hotfix] A couple of hotfixes

### PR DESCRIPTION
**Pull request by pnowojski**
## What is the purpose of the change

A couple of hotfixes

## Brief change log

Please check individual commit messages

## Verifying this change

Covered by existing tests and no functional changes.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? **(not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by StefanRRichter — APPROVED** [review]

File: flink-core/src/main/java/org/apache/flink/api/dag/Transformation.java [hunk 1]
```diff
@@ -638,29 +637,15 @@ public boolean equals(Object o) {
...
 
     @Override
     public int hashCode() {
-        int result = id;
-        result = 31 * result + name.hashCode();
-        result = 31 * result + (outputType != null ? outputType.hashCode() : 0);
-        result = 31 * result + parallelism;
-        result = 31 * result + (int) (bufferTimeout ^ (bufferTimeout >>> 32));
-        return result;
+        return Objects.hash(id, name, outputType, parallelism, bufferTimeout);
```
Comment (snuyanzin) [inline flink-core/src/main/java/org/apache/flink/api/dag/Transformation.java:649]: Not sure how critical it is here
however it seems `Objects#hash` for more than one arg could cost more
more details at https://bugs.openjdk.org/browse/JDK-8332840

Comment (pnowojski) [reply flink-core/src/main/java/org/apache/flink/api/dag/Transformation.java:649]: Thanks for pointing this out but I don't see how this should matter in our case. I have never seen any `Objects.hash` call anywhere being issue in Flink. Also `Transformation` is not used on any critical path, but during job submission, where there are tons of much heavier operations happening.

**Comment by pnowojski** [comment]
@flinkbot run azure

**Comment by pnowojski** [comment]
@flinkbot run azure

**Comment by pnowojski** [comment]
@flinkbot run azure

**Comment by pnowojski** [comment]
I can not get the azure to work for this PR, something has got stuck. Going to create a new one

**Pull request closed without merging**